Docker 🐳

Azmah-Bad opened the pull request:
Dockerized the API and all it's dependancies:
 - 2 mysql containers 
 - NodeJS container 
 - Keycloak container 🔑

Azmah-Bad committed:
[CI]: Added Dockerfiles 🐳 #2
+ Dockerfile to build the api's image
+ Docker-compose to run :
- NodeJS
- MySQL
- Keylcloak

Azmah-Bad committed:
[CI]: .dockerignore 🐳 #2
makes for faster loading and reloading

Azmah-Bad committed:
[CI]: 🐳 create a 2nd mysql container #2
now we have 2 mysql instances one handles the keycloak database
the other handles the api's database

Azmah-Bad committed:
[CI]: 🐳 API's container up and running #2
modified some environement variables to make the api able to fit in a container

Azmah-Bad committed:
[CI]: 🐳 API's container fix #2
- fixed an issue where the container wasn't getting environment variables.

Azmah-Bad committed:
[CI]: added docker 🐳 to documentation 📖 #2

Azmah-Bad committed:
[REFACTOR]: took sequelize setup out of index.js
this will make it easier to populate the DB and makes for greater readability and a lighter index.js file

Azmah-Bad committed:
[REFACTOR]: refactored populate.js 👨‍👩‍👦
Populate was a function that run before server was listing, Now you can run populate the database by running `node populate.js` or `npm populate`

Azmah-Bad committed:
[CI]: added some tests 🧪

Azmah-Bad committed:
[CI]: added notifications tests 🧪

Azmah-Bad committed:
[CI]: added activity_location tests 🧪

Azmah-Bad committed:
[REFACTOR]

Azmah-Bad committed:
[CI]: added mocha ☕️ as dev dependency

Azmah-Bad committed:
[CI]: 🔧 fixed the populate script
Made populate.js sync the database before populating it.

Azmah-Bad committed:
[CI]: 🕸 Added networks to the containers
greater isolation between the docker containers

titouan-joseph reviewed (approved):
J'ai test, ca marche ! :)

Azmah-Bad committed:
[CI]:🔨 docker runs on first try
- added a script to add the keycloak hostname on linux
- corrected the server hostname on keycloak.json
- made API restart on failure because it always crash on first try

Azmah-Bad committed:
[REFACTOR]:📁 moved setup files to setup directory

Azmah-Bad committed:
[DOC]: added the setup cmd in the setup documentation

Azmah-Bad committed:
Merge branch 'master' into Docker

Azmah-Bad committed:
[CI]: added create_admin.sh
a bash script to create a user with admin role in Keycloak
also the docker image doesn't contain dev dependencies

Azmah-Bad committed:
[CI]: 🛠fixed an issue causing the keycloak domain not to display correctly

Azmah-Bad committed:
[CI]: 📪 added postman collection for tests
CI can now use the postman collection to tests the API by running `npm run test`

Azmah-Bad committed:
[CI]: 📫 CI server runs postman collection

Azmah-Bad committed:
[POPULATION]: 👩‍👧 added depopulate.js
a script to depopulate the database for testing purposes

Azmah-Bad committed:
[DOC]: 📖 added useful scripts to README.md
added documentation about the useful scripts
